crates/services/gas_price_service/src/v0/tests.rs

```diff
@@ -212,14 +214,14 @@ async fn next__new_l2_block_saves_old_metadata() {
         algo_updater,
     );
 
-    // when
     let read_algo = service.next_block_algorithm();
-    let mut watcher = StateWatcher::default();
     let start = read_algo.next_gas_price();
+    let mut watcher = StateWatcher::started();
```

Comment:
is there a reason this isn't `StateWatcher::default()`?

Reply:
There were some places that `process_l2_block_res` wasn't getting called because of `get_l2_block` yielding a result, but because the service was shutting down (and we always check if there is an l2 block available before shutting down).

I added the `started` in a few places that it was shutting down early.

That being said, this isn't doing what the test says it's doing `next__new_l2_block_saves_old_metadata`. It's checking the gas price:
```rs
    // then
    let new = read_algo.next_gas_price();
    assert_ne!(start, new);
```

So I'll fix that.

Reply:
Fixed.